test/realtime/alerts_fetcher_test.exs

```diff
@@ -0,0 +1,218 @@
+defmodule Realtime.AlertsFetcherTest do
+  use ExUnit.Case
+  import Test.Support.Helpers
+  import ExUnit.CaptureLog
+  alias Realtime.AlertsFetcher
+
+  @api_response %{
+    "data" => [
+      %{
+        "attributes" => %{
+          "informed_entity" => [
+            %{"route" => "1"}
+          ],
+          "effect" => "DETOUR",
+          "service_effect" => "Route 1 detour"
+        },
+        "type" => "alert",
+        "id" => 123
+      }
+    ]
+  }
+
+  @complex_api_response %{
+    "data" => [
+      %{
+        "attributes" => %{
+          "informed_entity" => [
+            %{"route" => "1"},
+            %{"route" => "1"},
+            %{"route" => "2"}
+          ],
+          "effect" => "DETOUR",
+          "service_effect" => "Route 1 and 2 detour"
+        },
+        "type" => "alert",
+        "id" => 123
+      },
+      %{
+        "attributes" => %{
+          "informed_entity" => [
+            %{"route" => "2"}
+          ],
+          "effect" => "DETOUR",
+          "service_effect" => "Different route 2 detour"
+        },
+        "type" => "alert",
+        "id" => 456
+      },
+      %{
+        "attributes" => %{
+          "informed_entity" => [
+            %{"route" => "1"}
+          ],
+          "effect" => "UNKNOWN_EFFECT",
+          "service_effect" => "Non-detour alert"
+        },
+        "type" => "alert",
+        "id" => 789
+      }
+    ]
+  }
+
+  describe "start_link/1" do
+    test "starts GenServer" do
+      assert {:ok, _pid} = AlertsFetcher.start_link(update_fn: fn _ -> :ok end)
+    end
+  end
+
+  describe "init/1" do
+    test "sets state and queues up initial check" do
+      api_url = "https://test.com/"
+      api_key = "test_key"
+      reassign_env(:skate, :api_url, api_url)
+      reassign_env(:skate, :api_key, api_key)
+
+      update_fn = fn _ -> :ok end
+
+      assert {:ok,
+              %{
+                update_fn: ^update_fn,
+                poll_interval_ms: 500,
+                api_url: ^api_url,
+                api_key: ^api_key
+              },
+              {:continue, :initial_poll}} =
```

Comment:
comment(non-blocking): I'm new to elixir but this indentation/wrapping is kinda funky?

Because I'm learning, I'm wondering if the pattern match or assert be factored out to make it easier to read?

This seems like fairly standard elixir syntax so I don't think this needs to change.

Reply:
I see what you mean about how it's weird, but it's what the Elixir LSP server did for formatting and `mix format --check` likes it. I'm not sure what part you mean that we might be able to factor out - do you mean something like putting `{:continue, :initial_poll}` in its own variable and then pinning it in the match?